db/db_impl/db_impl.h

```diff
@@ -552,7 +552,15 @@ class DBImpl : public DB {
   using DB::StartTrace;
   virtual Status StartTrace(
       const TraceOptions& options,
-      std::unique_ptr<TraceWriter>&& trace_writer) override;
+      std::unique_ptr<TraceWriter>&& trace_writer) override {
+    std::list<WriteBatch*> initial_traced_contents;
+    return StartTrace(options, std::move(trace_writer),
+                      initial_traced_contents);
+  };
+
+  virtual Status StartTrace(
+      const TraceOptions& options, std::unique_ptr<TraceWriter>&& trace_writer,
+      const std::list<WriteBatch*>& initial_traced_contents);
```

Comment:
I chose it over vector just because I don't have to access it by indexing like initial_traced_contents[n] and a basic std::list suffice. Do you see a reason of not using list? I don't have strong preference here.

Reply:
Thanks for the clarification. I don't have a strong opinion here, and it's not public API, thus changing should be easy too.